sentry-android-core/src/main/java/io/sentry/android/core/util/RootChecker.java

```diff
@@ -0,0 +1,147 @@
+  }
+
+  /**
+   * Android Roms from Google are build with release-key tags. If test-keys are present, this can
+   * mean that the Android build on the device is either a developer build or an unofficial Google
+   * build.
+   *
+   * @return whether if it contains test keys or not
+   */
+  private static boolean checkTestKeys() {
+    final String buildTags = Build.TAGS;
+    return buildTags != null && buildTags.contains("test-keys");
+  }
+
+  /**
+   * Often the rooted device have the following files . This method will check whether the device is
+   * having these files or not
+   *
+   * @param logger the Logger
+   * @return whether if the root files exist or not
+   */
+  private static boolean checkRootFiles(final @NotNull ILogger logger) {
+    final String[] paths = {
+      "/system/app/Superuser.apk",
+      "/sbin/su",
+      "/system/bin/su",
+      "/system/xbin/su",
+      "/data/local/xbin/su",
+      "/data/local/bin/su",
+      "/system/sd/xbin/su",
+      "/system/bin/failsafe/su",
+      "/data/local/su",
+      "/su/bin/su",
+      "/su/bin",
+      "/system/xbin/daemonsu"
+    };
+
+    for (final String path : paths) {
+      try {
+        if (new File(path).exists()) {
+          return true;
+        }
+      } catch (Exception e) {
+        logger.log(
+            SentryLevel.ERROR, e, "Error when trying to check if root file %s exists.", path);
+      }
+    }
+    return false;
+  }
+
+  /**
+   * this will check if SU(Super User) exist or not
+   *
+   * @param logger the Logger
+   * @return whether su exists or not
+   */
+  private static boolean checkSUExist(final @NotNull ILogger logger) {
+    Process process = null;
+    final String[] su = {"/system/xbin/which", "su"};
+
+    try {
+      process = Runtime.getRuntime().exec(su);
+
+      try (final BufferedReader reader =
+          new BufferedReader(new InputStreamReader(process.getInputStream(), UTF_8))) {
+        return reader.readLine() != null;
+      }
+    } catch (Throwable e) {
+      logger.log(SentryLevel.ERROR, "Error when trying to check if SU exists.", e);
+      return false;
+    } finally {
+      if (process != null) {
+        process.destroy();
+      }
+    }
+  }
+
+  /**
+   * some application hide the root status of the android device. This will check for those files
+   *
+   * @param context the Context
+   * @return whether the root packages exist or not
+   */
+  private static boolean checkRootPackages(final @NotNull Context context) {
+    final PackageManager pm = context.getPackageManager();
+    if (pm != null) {
+      final String[] packages = {
+        "com.devadvance.rootcloak",
+        "com.devadvance.rootcloakplus",
+        "com.koushikdutta.superuser",
+        "com.thirdparty.superuser",
+        "eu.chainfire.supersu", // SuperSU
+        "com.noshufou.android.su" // superuser
+      };
+
+      for (final String pkg : packages) {
+        try {
+          pm.getPackageInfo(pkg, 0);
```

Comment:
`m`: Wouldn't `pm.getInstalledPackages(...);` be a better approach, because we avoid catching 6 exceptions.

Reply:
@philipphofmann I'd not like that an SDK checks all the installed packages of a device.
Also, this is arguable, what is more expensive? 6 exceptions or looping ~50 installed apps? can't say one or the other

Reply:
Checking all installed packages is an argument for me, but looping through the packages I think has to be faster, because you just have one call to the package manager and then you are done. The other approach has 6 calls to the package manager and 6 exceptions, I think this should be slower. I'm fine with both ways.

Reply:
> Checking all installed packages is an argument for me

@philipphofmann I don't get it, wdym?

Reply:
I mean that is an argument for keeping it like it is.